### PR TITLE
Extract implicitly ignored annotation names to separate class

### PR DESCRIPTION
**Pull request by Majkl578**
* To improve readability and maintainability of this list.
* Categorized, each category ordered alphabetically.
* Will be further used in Annotations 2.0.

### CODE REVIEW

File: lib/Doctrine/Annotations/ImplicitlyIgnoredAnnotationNames.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ *
+ *  All names are case sensitive.
+ */
+final class ImplicitlyIgnoredAnnotationNames
+{
+    private const Reserved = [
+        'Annotation' => true,
+        'Attribute'  => true,
+        'Attributes' => true,
+        /* Can we enable this? 'Enum' => true, */
```
Comment (SenseException) [inline lib/Doctrine/Annotations/ImplicitlyIgnoredAnnotationNames.php]: What about this question? This is from 5 years ago. I think it can be removed.

Comment (Majkl578) [reply lib/Doctrine/Annotations/ImplicitlyIgnoredAnnotationNames.php]: Didn't really want to do side-effects during just _extracting_ the annotations. Also in 2.0 these reserved annotations will likely not be needed anymore.

File: lib/Doctrine/Annotations/ImplicitlyIgnoredAnnotationNames.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        'codingStandardsIgnoreStart' => true,
+        'codingStandardsIgnoreEnd'   => true,
+    ];
+
+    private const SlevomatCodingStandard = [
+        'phpcsSuppress' => true,
+    ];
+
+    public const LIST = self::Reserved
+        + self::WidelyUsedNonStandard
```
Comment (Ocramius) [inline lib/Doctrine/Annotations/ImplicitlyIgnoredAnnotationNames.php]: Ooooooh, I like this style of merging stuff in constants!

**Comment by Ocramius** [comment]
This can go to `master` into `1.x` as-is.

**Comment by Ocramius** [comment]
As per discussion with @alcaeus and @Majkl578, `master` is already `2.0.0`, so this is going there.

**Pull request merged**